### PR TITLE
Allow different API Base for Navitia Providers

### PR DESCRIPTION
**Pull request by grote**
This is especially useful if you want to run a Navitia instance on your
own server.

### CODE REVIEW

File: enabler/src/de/schildbach/pte/AbstractNavitiaProvider.java [hunk 1]
```diff
@@ -77,7 +77,7 @@
 {
 	protected final static String SERVER_PRODUCT = "navitia";
 	protected final static String SERVER_VERSION = "v1";
-	protected static final String API_BASE = "http://api.navitia.io/" + SERVER_VERSION + "/";
+	protected static String API_BASE = "http://api.navitia.io/" + SERVER_VERSION + "/";
```
Comment (schildbach) [inline enabler/src/de/schildbach/pte/AbstractNavitiaProvider.java]: Can you make this a proper instance variable?

File: enabler/test/de/schildbach/pte/live/ItalyProviderLiveTest.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
 {
 	public ItalyProviderLiveTest()
 	{
-		super(new ItalyProvider(secretProperty("navitia.authorization")));
+		super(new ItalyProvider("https://api.navitia.io/v1/", secretProperty("navitia.authorization")));
```
Comment (schildbach) [inline enabler/test/de/schildbach/pte/live/ItalyProviderLiveTest.java]: This is already the default, no? Except for the s in https. If Navitia supports https, we should probably change the default.

**Comment by grote** [comment]
Thanks for the review! I updated this PR.

**Comment by schildbach** [comment]
I merged this with small changes. I left the HTTPS change for a future commit, actually I remembered we tried this a year ago with the ParisProvider and had some compatibility issues back then.

Thanks for your contribution!

**Pull request closed without merging**